java/ql/lib/semmle/code/java/security/CleartextStorageCookieQuery.qll

```diff
@@ -0,0 +1,48 @@
+/** Provides classes and predicates to reason about cleartext storage in cookies. */
+
+import java
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.DataFlow3
+import semmle.code.java.security.CleartextStorageQuery
+
+private class CookieCleartextStorageSink extends CleartextStorageSink {
+  CookieCleartextStorageSink() { this.asExpr() = cookieInput(_) }
+}
+
+/** The instantiation of a cookie, which can act as storage. */
+class Cookie extends Storable, ClassInstanceExpr {
+  Cookie() {
+    this.getConstructor().getDeclaringType().getQualifiedName() = "javax.servlet.http.Cookie"
```

Comment:
For getting a specific type `hasQualifiedName` generally performs better than `getQualifiedName`, since the latter needs to scan all types to construct the concatenated name whereas the former can do a direct lookup.
```suggestion
    this.getConstructor().getDeclaringType().hasQualifiedName("javax.servlet.http", "Cookie")
```